tictactoe_env.py

```diff
@@ -0,0 +1,104 @@
+import numpy as np
+
+
+class TicTacToeEnv(object):
+    def __init__(self):
+        """
+        state space: 2x3x3
+        action space: 18 possible values -- 1 hot
+            action: make actions a single integer -- in range(0, 18)
+        assume x's always go first
+        """
+        self.state = np.zeros((2, 3, 3))
+        self.action_dims = (2, 3, 3)
+
+    # The following 4 methods are called outside of the environment
+    def get_next_state(self, state, action_int):
+        # turn = get_turn_tictactoe(state)
+        # if turn
+        action_array = self.convert_int_to_action(action_int)
+        next_state = state + action_array
+        return next_state
+
+    def get_legal_actions(self, state):
+        turn = self.get_turn_tictactoe(state)
+        # legal_actions = np.zeros((2, 3, 3), dtype=int)
+        legal_actions = []
+        for i in range(3):
+            for j in range(3):
+                if state[:, i, j].sum() == 0:
+                    action_array = np.zeros((2, 3, 3), dtype=int)
+                    action_array[turn, i, j] = 1
+                    action_int = self.convert_action_to_int(action_array)
+                    legal_actions.append(action_int)
+        return np.array(legal_actions)
+
+    def is_game_over(self, state):
+        result = self.outcome(state)
+        if result == 2:
+            return 0
+        else:
+            return 1
+
+    def outcome(self, state):
```

Comment:
So only x can win in this environment? Shouldn't this return `-1` sometimes?

Reply:
omg you are totally right.  I'll fix.